ra/ra.go

```diff
@@ -760,14 +762,27 @@ func (ra *RegistrationAuthorityImpl) checkAuthorizationsCAA(
 // rechecked because their associated authorizations are sufficiently old and
 // performs the CAA checks required for each. If any of the rechecks fail an
 // error is returned.
-func (ra *RegistrationAuthorityImpl) recheckCAA(ctx context.Context, names []string) error {
+func (ra *RegistrationAuthorityImpl) recheckCAA(ctx context.Context, authzs []*core.Authorization) error {
 	ra.stats.Inc("recheck_caa", 1)
-	ra.stats.Inc("recheck_caa_names", int64(len(names)))
-	ch := make(chan error, len(names))
-	for _, name := range names {
-		go func(name string) {
+	ra.stats.Inc("recheck_caa_authzs", int64(len(authzs)))
+	ch := make(chan error, len(authzs))
+	for _, authz := range authzs {
+		go func(authz *core.Authorization) {
+			name := authz.Identifier.Value
+			var method string
+			for _, challenge := range authz.Challenges {
```

Comment:
I think its worth dropping a short comment here to highlight the baked in assumption that an authorization is only ever set to status valid based on a single challenge being valid.

Reply:
Done - presuming that we can have multiple valid challenges, with differing types - obviously, this will currently just recheck CAA with the first valid method... not sure if we need to handle the case where, for example both DNS-01 and HTTP-01 have been validated previously, however CAA now only allows one of those (seems like a rather obscure corner case).